### PR TITLE
Run e2e tests in both Chrome and Electron

### LINKED ISSUE #5357: Run e2e tests in all browsers by default
### Current behavior:

e2e tests only run on 1 browser at a time. To run on multiple browsers requires multiple e2e tests.

### Desired behavior:

* Run an `electron-e2e-tests` stage and a `chrome-e2e-tests` stage that set `E2E_BROWSER=electron/chrome` respectively
* For each e2e test:
	* If a `browser` field is not specified...
		* Run the test against all browsers individually
	* If `browser` is specified and a string, run the test in that browser individually
	* If `browser` is specified and an array, run the test in those browsers individually
* When running a test against an individual browser:
	* If E2E_BROWSER is set and is not the current browser, bail since the current test is not meant for the current stage
* By default, only 1 snapshot will be saved per test, and all browsers should conform to that snapshot
	* can be overridden with `snapshotPerBrowser: true`

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #5357 

### User facing changelog

N/A - not user facing

### Additional details

- [x] Pending #4628, since rn chrome snapshots are different, they contain the "no video" warning

### PR Tasks

- [ ] Have tests been added/updated?

### CODE REVIEW

**Comment by bahmutov** [comment]
I have opened https://github.com/bahmutov/snap-shot-it/issues/443 to allow multiple tests to "share" a named snapshot

**Comment by bahmutov** [comment]
I have published `snap-shot-it@7.9.0` which implements shared named snapshots, see https://github.com/bahmutov/snap-shot-it/commit/cad8ad6403a0b6740ebd1d65fd22876ee2aeef99

```
snapshot('my shared snapshot', value, { allowSharedSnapshot : true })
```

File: circle.yml [hunk 1]
```diff
@@ -37,6 +37,44 @@ executors:
...
+  run-e2e-tests-electron:
+    parameters:
+      chunk:
+        description: e2e test chunk number
+        type: integer
+    steps:
+      - run-e2e-test:
+          browser: electron
+          chunk: << parameters.chunk >>
+
```
Comment (brian-mann) [inline circle.yml]: I think this is too much abstraction and would just prefer to use the base `run-e2e-test` command and pass in `browser` as an option.

File: packages/server/__snapshots__/0_simple_spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+exports['e2e simple stuff / passes'] = `
```
Comment (brian-mann) [inline packages/server/__snapshots__/0_simple_spec.js]: you can remove this file / snapshot. it was just an example

File: packages/server/test/e2e/0_simple_spec.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const e2e = require('../support/helpers/e2e')
```
Comment (brian-mann) [inline packages/server/test/e2e/0_simple_spec.js]: remove this file

**Pull request merged**